DigitalLearningSolutions.Data/DataServices/CourseDelegatesDataService.cs

```diff
@@ -12,21 +12,21 @@ public interface ICourseDelegatesDataService
 
     public class CourseDelegatesDataService : ICourseDelegatesDataService
     {
-        private readonly IDbConnection connection;
-
         private const string AllAttemptsQuery =
             @"(SELECT COUNT(aa.AssessAttemptID)
                 FROM dbo.AssessAttempts AS aa
                 INNER JOIN dbo.Candidates AS can ON can.CandidateID = aa.CandidateID
                 WHERE aa.CustomisationID = cu.CustomisationID AND aa.[Status] IS NOT NULL
-                AND can.CentreID = @centreId) AS AllAttempts";
+                AND can.CentreID = @centreId AND can.CandidateId = c.CandidateId) AS AllAttempts";
 
         private const string AttemptsPassedQuery =
             @"(SELECT COUNT(aa.AssessAttemptID)
                 FROM dbo.AssessAttempts AS aa
                 INNER JOIN dbo.Candidates AS can ON can.CandidateID = aa.CandidateID
                 WHERE aa.CustomisationID = cu.CustomisationID AND aa.[Status] = 1
-                AND can.CentreID = @centreId) AS AttemptsPassed";
+                AND can.CentreID = @centreId AND can.CandidateId = c.CandidateId) AS AttemptsPassed";

```

Comment:
The addition of the check on candidate number on these query strings is not part of the requirements for this ticket, but it fixes a minor issue I saw while testing this. 

Before the user reaches the new Delegate Progress page, they need to click a button on the Course Delegates expandable which also shows a pass rate. That pass rate was being calculated incorrectly as these values were being recovered incorrectly. These queries used to get all the attempts for all users on that course, rather than the individual in question. The additional check here means only the numbers for the individual are obtained, so the values are correct.

Reply:
Good catch. It worries me a bit that we're relying on the `Candidates` table in a different query being aliased as `c`.
What about changing these into methods and passing the candidate id field reference in?
Then adding a mention of the candidate in the name makes calls read slightly more sensibly.
So e.g. `QueryForNumAttemptsPassedByCandidate(string candidateIdFieldReference)`.

Reply:
These are only being used on one method, and are separated into their own consts just to make the main query a bit easier to read and understand. I've pulled these definitions into the method to help restrict any of problems that might occur in future with the candidate references.